src/System.Linq/tests/ConcatTests.cs

```diff
@@ -251,20 +251,34 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             var supposedlyLargeCollection = new DelegateBasedCollection<int> { CountWorker = () => int.MaxValue };
             var tinyCollection = new DelegateBasedCollection<int> { CountWorker = () => 1 };
 
+            Action<Action> assertThrows = (testCode) =>
+            {
+                // The full .NET framework uses unsigned arithmetic summing up collection counts.
+                // See https://github.com/dotnet/corefx/pull/6131.
+                if (PlatformDetection.IsFullFramework)
+                {
+                    testCode();
+                }
+                else
+                {
+                    Assert.Throws<OverflowException>(testCode);
```

Comment:
So we're throwing in .NET Core but not in desktop?  We should revisit that decision to see if it's really the right one.